rd-cpp/src/rd_framework_cpp/src/main/wire/SocketWire.cpp

```diff
@@ -552,55 +561,53 @@ SocketWire::Server::Server(Lifetime lifetime, IScheduler* scheduler, uint16_t po
 
-		logger->info("{}: started, port: {}.", this->id, this->port);
-
-		try
+		while (!lifetime->is_terminated())
 		{
-			while (!lifetime->is_terminated())
+			try
 			{
-				try
-				{
-					logger->info("{}: accepting started", this->id);
-					CActiveSocket* accepted = ss->Accept();
-					RD_ASSERT_THROW_MSG(
-						accepted != nullptr, fmt::format("{}: accepting failed, reason: {}", this->id, ss->DescribeError()));
-					socket.reset(accepted);
-					logger->info("{}: accepted passive socket {}/{}", this->id, socket->GetClientAddr(), socket->GetClientPort());
-					RD_ASSERT_THROW_MSG(socket->DisableNagleAlgoritm(),
-						fmt::format("{}: tcpNoDelay failed, reason: {}", this->id, socket->DescribeError()));
+				logger->info("{}: accepting started", this->id);
+				
+				// [HACK]: Fix RIDER-51111.
+				// winsock blocking accept hangs after creating new process with createprocess with inheritHandles=true
+				// property. Unreal Engine uses the same logic for handling sockets where they wait for timeout on select
+				// before trying to accept connection.
+				while(ss->IsSocketValid() && !ss->Select(0, 300)){}
+				
+				CActiveSocket* accepted = ss->Accept();
+				RD_ASSERT_THROW_MSG(
+					accepted != nullptr, fmt::format("{}: accepting failed, reason: {}", this->id, ss->DescribeError()));
+				socket.reset(accepted);
+				logger->info("{}: accepted passive socket {}/{}", this->id, socket->GetClientAddr(), socket->GetClientPort());
+				RD_ASSERT_THROW_MSG(socket->DisableNagleAlgoritm(),
+					fmt::format("{}: tcpNoDelay failed, reason: {}", this->id, socket->DescribeError()));
 
+				{
+					std::lock_guard<decltype(lock)> guard(lock);
+					if (lifetime->is_terminated())
 					{
-						std::lock_guard<decltype(lock)> guard(lock);
-						if (lifetime->is_terminated())
+						logger->debug("{}: closing passive socket", this->id);
+						if (!socket->Close())
 						{
-							logger->debug("{}: closing passive socket", this->id);
-							if (!socket->Close())
-							{
-								logger->error("{}: failed to close socket", this->id);
-							}
-							logger->info("{}: close passive socket", this->id);
+							logger->error("{}: failed to close socket", this->id);
 						}
+						logger->info("{}: close passive socket", this->id);
 					}
-
-					logger->debug("{}: setting socket provider", this->id);
-					set_socket_provider(socket);
-				}
-				catch (std::exception const& e)
-				{
-					logger->info("{}: closed with exception: {}", this->id, e.what());
 				}
+
+				logger->debug("{}: setting socket provider", this->id);
+				set_socket_provider(socket);
+			}
+			catch (std::exception const& e)
+			{
+				logger->info("{}: closed with exception: {}", this->id, e.what());
 			}
 		}
-		catch (std::exception const& e)
-		{
-			logger->error("{}: terminal socket error ({}).", this->id, e.what());
-		}
-
-		logger->info("{}: terminated, port: {}.", this->id, this->port);
```

Comment:
Let's have this "terminated+port" logging statement, and delete the unclear one `logger->debug("{}: thread expired", this->id)`.